src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,228 @@
+     *
+     * @param file The file (lecture pdf) to be split
+     * @return The prepared map
+     */
+    private Outline separateIntoUnits(MultipartFile file) throws IOException {
+        try (PDDocument document = PDDocument.load(file.getBytes())) {
+            Map<Integer, LectureUnitSplit> outlineMap = new HashMap<>();
+            Splitter pdfSplitter = new Splitter();
+            PDFTextStripper pdfStripper = new PDFTextStripper();
+            // split the document into single pages
+            List<PDDocument> pages = pdfSplitter.split(document);
+            int numberOfPages = document.getNumberOfPages();
+            ListIterator<PDDocument> iterator = pages.listIterator();
+
+            int outlineCount = 0;
+            int index = 1;
+            while (iterator.hasNext()) {
+                PDDocument currentPage = iterator.next();
+                String slideText = pdfStripper.getText(currentPage);
+
+                if (isOutlineSlide(slideText)) {
+                    outlineCount++;
+                    String[] lines = slideText.split("\r\n|\r|\n");
+
+                    // if it's the outline slide it will get the next bullet point as unit name.
+                    String unitName = lines[outlineCount + 1].replaceAll("[^a-zA-Z0-9\\s()_-]", "").replaceFirst("^\\s*", "");
+                    outlineMap.put(outlineCount, new LectureUnitSplit(unitName, outlineCount == 1 ? 1 : index, numberOfPages));
+
+                    if (outlineCount > 1) {
+                        updatePreviousUnitEndPage(outlineCount, outlineMap, index);
+                    }
+                }
+                currentPage.close(); // make sure to close the document
+                index++;
+            }
+            document.close(); // make sure to close the document
+            return new Outline(outlineMap, numberOfPages);
+        }
+    }
+
+    /**
+     * This method updates previous unit end page, and it's called if outline count is bigger then 1
+     *
+     * @param outlineCount Outline count is the number of slides that contain Outline
+     * @param outlineMap   Outline map with unit information
+     * @param index        index that shows current page
+     */
+    private void updatePreviousUnitEndPage(int outlineCount, @NotNull Map<Integer, LectureUnitSplit> outlineMap, int index) {
+        int previousOutlineCount = outlineCount - 1;
+        int previousStart = outlineMap.get(previousOutlineCount).startPage;
+        String previousUnitName = outlineMap.get(previousOutlineCount).unitName;
+        outlineMap.put(previousOutlineCount, new LectureUnitSplit(previousUnitName, previousStart, index - 1));
+    }
+
+    private boolean isOutlineSlide(final String slideText) {
+        return slideText.contains("Outline");
+    }
+
+    private record LectureUnitSplit(String unitName, int startPage, int endPage) {
+    }
+
+    private record Outline(Map<Integer, LectureUnitSplit> splits, int totalPages) {
```

Comment:
```suggestion
    /**
     * Maps from ??? to the split for all parts of the lecture unit.
     */
    private record Outline(Map<Integer, LectureUnitSplit> splits, int totalPages) {
```
What does the map key mean?

Reply:
The key is the unit number.